core/pkg/evaluator/fractional.go

```diff
@@ -72,13 +72,13 @@ func parseFractionalEvaluationData(values, data any) (string, []fractionalEvalua
 	return bucketBy, feDistributions, nil
 }
 
-func parseFractionalEvaluationDistributions(values []any) ([]fractionalEvaluationDistribution, error) {
+func parseFractionalEvaluationDistributions(values []interface{}) ([]fractionalEvaluationDistribution, error) {
```

Comment:
Unfortunately, this does not solve the underlying issue . `any` & `interface{}` are type aliasses.

You have to validate the content of the array and ignore nil values to continue.

Reply:
if values[i] == nil {
			continue
		}


Will this be good if I included it inside the loop.

Reply:
That's an option, but I would avoid adding the check into the `parseFractionalEvaluationDistributions`. The parameters here should already be filtered and validated.

The fix should focus on parsing of the distributions [1]. If the first parameter is not a string, then we should check it's non-nil 

[1] - https://github.com/open-feature/flagd/blob/core/v0.9.0/core/pkg/evaluator/fractional.go#L58-L64